index.js

```diff
@@ -7,12 +7,21 @@ function shift (stream) {
 }
 
 function getStateLength (state) {
+  if (state.readableLength != null) {
+    return state.readableLength
+  }
+
+  if (state.length != null) {
+    return state.lenght
```

Comment:
typo here

Reply:
The two lengths here refer to the next chunk in the buffer?

Reply:
That was wrong. Updated.